packages/cxl-ui/src/components/cxl-jw-player/mixins/BaseMixin.js

```diff
@@ -0,0 +1,235 @@
+import * as jose from 'jose';
+import { render } from 'lit';
+import { property } from 'lit/decorators.js';
+import { throttle } from 'lodash-es';
+import { parseSync } from 'subtitle';
+import { MD5 } from 'crypto-js';
+import { MediaQueryController } from '@vaadin/component-base/src/media-query-controller.js';
+
+export function BaseMixin(BaseClass) {
+  class Mixin extends BaseClass {
+    _boundOnTimeListener;
+
+    _chapters;
+
+    _jwPlayer;
+
+    _jwPlayerContainer;
+
+    // Device Detector media query.
+    _wideMediaQuery = '(min-width: 750px)';
+
+    @property({ attribute: 'api-secret', type: String }) apiSecret = '';
+
+    @property({ attribute: 'is-public', type: Boolean }) isPublic;
+
+    @property({ attribute: 'library-id', type: String }) libraryId;
+
+    @property({ attribute: 'library-source', type: String }) librarySource;
+
+    @property({ attribute: 'media-id', type: String }) mediaId;
+
+    @property({ attribute: 'media-source', type: String }) mediaSource;
+
+    @property({ attribute: 'playlist-id', type: String }) playlistId;
+
+    @property({ attribute: 'playlist-source', type: String }) playlistSource;
+
+    // MediaQueryController.
+    @property({ type: Boolean, reflect: true })
+    wide;
+
+    constructor() {
+      super();
+
+      this.addController(
+        new MediaQueryController(this._wideMediaQuery, (matches) => {
+          this.wide = matches;
+        })
+      );
+    }
+
+    async firstUpdated(_changedProperties) {
+      await super.firstUpdated(_changedProperties);
+
+      await this._beforeSetup();
+      this._setup();
+    }
+
+    updated(_changedProperties) {
+      super.updated(_changedProperties);
+      if (_changedProperties.has('captions') || _changedProperties.has('mediaId')) {
+        // this._setup();
+      }
+    }
+
+    get _scriptUrl() {
+      if (!this.libraryId && !this.librarySource) return false;
+
+      let scriptUrl;
+
+      if (this.libraryId) {
+        if (this.isPublic) {
+          scriptUrl = `https://content.jwplatform.com/libraries/${this.libraryId}.js`;
```

Comment:
Other places use `https://cdn.jwplayer.com` URL, shouldn't we use it here too?

Reply:
I can refactor this.